### PR TITLE
Extended Data Read and Execute Channel

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
- Add function to read extended data after receiving a WS_EXTDATA error.

- Add example remote execute on client side

```
./examples/client/client -p 22 -u pi2 -h <ip address> -c ./a.out
```

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Am working more on how the example client handles IO from stdin and stdout (i.e. when to echo to terminal).

**Comment by JacobBarthelmeh** [comment]
Adds the API
*) wolfSSH_stream_peek
*) wolfSSH_extended_data_read
*) wolfSSH_ConvertConsole

Adds examples/client/client options
*) -t (psuedo-terminal)
*) -R (raw mode with windows)
*) -c <string> (execute command)

Adds the build flag
*) --enable-term (on by default)

File: src/ssh.c [hunk 1]
```diff
@@ -786,6 +823,26 @@ int wolfSSH_stream_exit(WOLFSSH* ssh, int status)
...
+    byte*  buf;
+    word32 bufSz;
+
+    if (ssh == NULL || out == NULL) {
+        return WS_BAD_ARGUMENT;
+    }
+
+    bufSz = min(outSz, ssh->extDataBuffer.length - ssh->extDataBuffer.idx);
+    buf = ssh->extDataBuffer.buffer + ssh->extDataBuffer.idx;
+    WMEMCPY(out, buf, bufSz);
```
Comment (JacobBarthelmeh) [inline src/ssh.c]: probably should advance idx here so can drain buffer with reads

**Comment by ejohnstown** [comment]
File examples/sftpclient/sftpclient.c: lines 145 and 147, one is a duplicate. Looks like rebase debris.

File: examples/sftpclient/sftpclient.c [hunk 2]
```diff
@@ -144,6 +144,7 @@ static void clean_path(char* path)
             int enIdx = 0; /* end of cut */
             sz = (long)WSTRLEN(path);
 
+            sz = (long)WSTRLEN(path);
```
Comment (ejohnstown) [inline examples/sftpclient/sftpclient.c]: This is the duplicate.

**Pull request merged**